src/Altinn.Notifications.Core/Services/EmailOrderProcessingService.cs

```diff
@@ -15,41 +15,68 @@ public class EmailOrderProcessingService : IEmailOrderProcessingService
 {
     private readonly IEmailNotificationRepository _emailNotificationRepository;
     private readonly IEmailNotificationService _emailService;
+    private readonly IContactPointService _contactPointService;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="OrderProcessingService"/> class.
     /// </summary>
     public EmailOrderProcessingService(
         IEmailNotificationRepository emailNotificationRepository,
-        IEmailNotificationService emailService)
+        IEmailNotificationService emailService,
+        IContactPointService contactPointService)
     {
         _emailNotificationRepository = emailNotificationRepository;
         _emailService = emailService;
+        _contactPointService = contactPointService;
     }
 
     /// <inheritdoc/>
     public async Task ProcessOrder(NotificationOrder order)
     {
-        foreach (Recipient recipient in order.Recipients)
+        var recipients = order.Recipients;
+        var recipientsWithoutEmail = recipients.Where(r => !r.AddressInfo.Exists(ap => ap.AddressType == AddressType.Email)).ToList();
+
+        List<Recipient> agumentedRecipients = await _contactPointService.GetEmailContactPoints(recipientsWithoutEmail);
+
+        var augmentedRecipientDictionary = agumentedRecipients.ToDictionary(ar => $"{ar.NationalIdentityNumber}-{ar.OrganisationNumber}");
+
+        foreach (Recipient originalRecipient in recipients)
         {
-            await _emailService.CreateNotification(order.Id, order.RequestedSendTime, recipient);
+            if (augmentedRecipientDictionary.TryGetValue($"{originalRecipient.NationalIdentityNumber}-{originalRecipient.OrganisationNumber}", out Recipient? augmentedRecipient))
+            {
+                originalRecipient.AddressInfo.AddRange(augmentedRecipient!.AddressInfo);
```

Comment:
I suspect that AddressInfo were already added by the GetEmailContantPoints method. It's looping through the original list and making changes to elements in the list rather than making new recipients.